### ISSUE TITLE
Mysql error when installing Magento 2.4 by composer.

### ISSUE THREAD

**Issue by mtjweb** [magento/magento2#29680]
### Preconditions (*)

1. Magento 2.4
2. CentOS 7
3. Php 7.3.21
4. MariaDB 10.3.24

### Steps to reproduce (*)

1. In composer run the following command:
php bin/magento setup:install --base-url="http://example.com/" --db-host="localhost" --db-name="dbname" --db-user="dbuser" --db-password="dbpassword" --admin-firstname="admin" --admin-lastname="admin" --admin-email="info@example.com" --admin-user="admin" --admin-password="Password" --language="nl_NL" --currency="EUR" --timezone="Europe/Amsterdam" --use-rewrites="1" --backend-frontname="admin"

### Expected result (*)

1. Installation progressing without errors

### Actual result (*)

After step "Progress 4 / 1206" I get the following two errors
SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The maximum row size for the used table type, not counting BLOBs, is 8126. This includes storage overhead, check the manual. You have to change some columns
   to TEXT or BLOBs, query was: ALTER TABLE `inventory_source` ADD COLUMN `is_pickup_location_active` BOOLEAN NOT NULL DEFAULT 0 , ADD COLUMN `frontend_name` varchar(255) NULL DEFAULT "" , ADD COLUMN `frontend_description` text NULL

SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The maximum row size for the used table type, not counting BLOBs, is 8126. This includes storage overhead, check the manual. You have to change some columns
   to TEXT or BLOBs

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [X] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by mtjweb** [comment]
Problem was solved by adding the following code to my.cnf and restarting MySQL-server:

`# Important: inside the mysqld block
[mysqld]
# Add new log file size
innodb_log_file_size=512M
# Disable strict mode
innodb_strict_mode=0`

**Comment by floorz** [comment]
@mtjweb So was that MySQL config change the solution to this problem or just a means to silence MySQL. I am currently getting the same error during a deployment of Magento 2.4 to a staging server for a client.

I would think MySQL is complaining about the row size limit for a reason. I'm curious if changing those configs are going to allow for some kind of truncated data or side effects?

@magento-engcom-team Thoughts? Do you have knowledge to share here?

**Comment by mtjweb** [comment]
@floorz For me it resolved the problem and had no side effects or other problems.

**Comment by Dren7755** [comment]
I have the same issue during upgrade from M2.3.3 to M2.4.1
DB: 10.4.17-MariaDB-1

**Comment by Dren7755** [comment]
I have the same problem during upgrade from Magento 2.3.3 to 2.4.1

The main cause of a problem is my MariaDB v10.4.

Previously we have MariaDB 10.1.
We can see in official docs https://mariadb.com/kb/en/innodb-row-formats-overview/ that the default row format for 10.1 is COMPACT.
At the same time the default row format for 10.4 is DYNAMIC.
When we manually changed row format for `inventory_source` which is triggered a problem - it helped.

So I have changed row format to DYNAMIC for all magento's tables
